copy/copy.go

```diff
@@ -44,6 +46,7 @@ type imageCopier struct {
 	reportWriter      io.Writer
 	progressInterval  time.Duration
 	progress          chan types.ProgressProperties
+	mu                sync.RWMutex
```

Comment:
It would be nice to explicitly document which members are expected to be read-only throughout, and which members are protected by the mutex.  (Or, for now, we could just use an `x/sync/syncmap` for `cachedDiffIDs` if that’s the only protected member—but I guess that won’t last.)